modules/apps/petra/petra-url-pattern-mapper/src/main/java/com/liferay/petra/url/pattern/mapper/internal/StaticSizeTrieURLPatternMapper.java

```diff
@@ -53,6 +54,87 @@ public StaticSizeTrieURLPatternMapper(Map<String, T> values) {
 		}
 	}
 
+	@Override
```

Comment:
@arthurchan35... with this approach we are removing the long bitmask that accumulates the results, aren't we?

if I understand correctly this would decrease performance if our analysis was correct before, because we would be flushing the trie out of the processor cache when invoking the consumer. I believe it should perform better if we use the bitmask to collect while we are matching and iterate in the end to invoke the cosumer.